src/TimeSeries/encodings/timeseriespreprocessing.jl

```diff
@@ -0,0 +1,77 @@
+struct TSStats{T}
+    means::AbstractArray{T,2}
+    stds::AbstractArray{T,2}
+end
+
+function TSStats(means, stds)
+    TSStats{eltype(means)}(means, stds)
+end
+
+"""
+    TSPreprocessing() <: Encoding
+
+Encodes 'TimeSeriesRow's by normalizing the time-series values. The time-series can
+either be normalized by each variable or time-step.
+
+Encodes
+- 'TimeSeriesRow' -> 'TimeSeriesRow'
+"""
+
+struct TSPreprocessing <: Encoding
+    tfms
+    stats::TSStats
+end
+
+function TSPreprocessing(stats::TSStats)
+    base_tfms = [
+        TSStandardize
+    ]
+    return TSPreprocessing{}(base_tfms, stats)
+end
+
+function TSStandardize(
+    obs,
+    p::TSPreprocessing
+)
+    means = p.stats.means
+    stds  = p.stats.stds
+    obs = obs .- means
+    obs = obs ./ stds
+    return obs
+end
+
+function tsdatasetstats(
+    data;
+    by_var=false,
+    by_step=false
+)
+    drop_axes = []
+    if (by_var)
+        append!(drop_axes,2)
+    else
+        append!(drop_axes,3)
+    end 
+    axes = [ax for ax in [1, 2, 3] if !(ax in drop_axes)]
+    mean = Statistics.mean(data.table, dims=axes)
+    std  = Statistics.std(data.table, dims=axes)
```

Comment:
```suggestion
    std  = Statistics.std(data.table, dims=axes, mean=mean)
```
If the data source supports it, this is more efficient.

Reply:
Yes, I tried using `Statistics.stdm(itr, mean)` earlier, couldn't seem to get it working. Will look into it further and keep this comment open till then.